### PR TITLE
converts old flow type declarations to d.ts files

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

converts old flow declaration files to typescript ones

they have been pulled over from `javascripts.flow.archive` and automatically converted with https://flow-to-ts.netlify.app.

no promises that the types are correct  – they are just whatever they were before, plus any lint fixes (mainly swapping `{}` for `unknown`)

### CODE REVIEW

**Review by gtrufitt — APPROVED** [review]
Wonderful

File: static/src/javascripts/types/user-attributes.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//the type of the response returned by https://members-data-api.theguardian.com/user-attributes/me/mma-monthlycontribution
```
Comment (mxdvl) [inline static/src/javascripts/types/user-attributes.d.ts:1]: This link does not work

File: static/src/javascripts/types/discussion-profile.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// model: https://github.com/guardian/discussion-api/blob/master/discussion-api/src/main/scala/com.gu.discussion.api/model/Profile.scala#L26
```
Comment (mxdvl) [inline static/src/javascripts/types/discussion-profile.d.ts:1]: Should we pin this to a permalink?

File: static/src/javascripts/types/ab-tests.d.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
+import type { ReminderFields } from 'common/modules/commercial/templates/acquisitions-epic-reminder';
+import type { TickerSettings } from 'common/modules/commercial/ticker';
```
Comment (mxdvl) [inline static/src/javascripts/types/ab-tests.d.ts:2]: Don’t these need to become relative?

**Review by mxdvl — COMMENTED** [review]
Just small comments, but this looks really promising!

File: static/src/javascripts/types/user-attributes.d.ts (hunk 1 above)
Comment (sndrs) [reply static/src/javascripts/types/user-attributes.d.ts:1]: yeah I'm not really checking these files, much of the content I've had nothing to do with, def more of a conversion process than anything else

File: static/src/javascripts/types/ab-tests.d.ts (hunk 3 above)
Comment (sndrs) [reply static/src/javascripts/types/ab-tests.d.ts:2]: the files don't seem to exist, but I don't really want to get into changing the contents in this PR?

**Review by mxdvl — APPROVED** [review]

**Pull request merged**